test/tdf-roundtrips.bats

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env bats
+
+# Tests for creating and reading TDF files with various settings
+# Notably, tests both 'ztdf' and 'nano' formats.
+
+@test "examples: roundtrip Z-TDF" {
+  echo "[INFO] create a tdf3 format file"
+  run go run ./examples encrypt "Hello Zero Trust"
+  echo "[INFO] echoing output; if successful, this is just the manifest"
+  echo "$output"
+
+  echo "[INFO] Validate the manifest lists the expected kid in its KAO"
+  kid=$(jq -r '.encryptionInformation.keyAccess[0].kid' <<<"${output}")
+  echo "$kid"
+  [ $kid = r1 ]
+
+  echo "[INFO] decrypting..."
+  run go run ./examples decrypt sensitive.txt.tdf
+  echo "$output"
+  printf '%s\n' "$output" | grep "Hello Zero Trust"
+}
+
+@test "examples: roundtrip nanoTDF" {
+  echo "[INFO] creating nanotdf file"
+  go run ./examples encrypt -o sensitive.txt.ntdf --nano "Hello NanoTDF"
+
+  echo "[INFO] decrypting nanotdf..."
+  go run ./examples decrypt sensitive.txt.ntdf
+  go run ./examples decrypt sensitive.txt.ntdf | grep "Hello NanoTDF"
+}
+
+@test "examples: legacy key support Z-TDF" {
+  echo "[INFO] validating default key is r1"
+  [ $(grpcurl -plaintext "localhost:8080" "kas.AccessService/PublicKey" | jq -e -r .kid) = r1 ]
+
+  echo "[INFO] encrypting samples"
+  go run ./examples encrypt -o sensitive-with-no-kid.txt.tdf --no-kid-in-kao "Hello Legacy"
+  go run ./examples encrypt -o sensitive-with-kid.txt.tdf "Hello with Key Identifier"
+
+  echo "[INFO] decrypting..."
+  go run ./examples decrypt sensitive-with-no-kid.txt.tdf | grep "Hello Legacy"
```

Comment:
should we use the bats-assert helper here?

Reply:
Pro: Simpler test assertions and clearer errors make easier debugging.
Con: Another dep and thing to learn. Hard to install on macos

Moved to another PR for discussion:

https://github.com/opentdf/platform/pull/981/commits/82ce1f5a35b6640e994806daf450eb5465fd8fd8